src/main/java/constants/Regex.java

```diff
@@ -63,8 +60,10 @@ private static Pattern generateExtractorPattern(String includedOptionFlag, Strin
         {
             regex.append("(?<=\\s)");                                       // [any of the excludedOptionFlags
             regex.append("-[");
-            for (String s : excludedOptionFlags) {
-                regex.append(s.replace("-", EMPTY_REGEX));
+            for (String option : OPTIONS_ARRAY) {
+                if (!option.equals(includedOptionFlag)) {
+                    regex.append(option.replace("-", EMPTY_REGEX));
+                }
```

Comment:
I will preserve this current setup for now because it makes sense to terminate the search on all valid flags. We can look into developing more specific methods that take in specific flags to exclude if/when necessary.